src/common/snippets/src/lowered/pass/init_buffers_default.cpp

```diff
@@ -17,17 +16,22 @@ namespace pass {
 bool InitBuffersDefault::run(lowered::LinearIR& linear_ir, lowered::LinearIR::constExprIt begin, lowered::LinearIR::constExprIt end) {
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::InitBuffersDefault");
 
-    size_t id = 0;
+    size_t idx = 0;
     size_t offset = 0;
     for (auto expr_it = begin; expr_it != end; ++expr_it) {
         const auto& expr = *expr_it;
         const auto op = expr->get_node();
         if (const auto buffer = ov::as_type_ptr<op::Buffer>(op)) {
-            AllocateBuffers::set_buffer_offset(expr, offset);
-            buffer->set_id(id);
-
-            offset += buffer->get_byte_size();
-            id++;
+            buffer->set_reg_group(idx);
+            buffer->set_cluster_id(idx);
+
+            if (!buffer->is_defined()) {
+                buffer->set_offset(utils::get_dynamic_value<size_t>());
```

Comment:
As I understood, if the buffer is not defined, its offset is already dynamic. Why do we need explicitly set dynamic value here?

Reply:
Just in case 😃 To see which offset is set in this default pass for dynamic buffers. Otherwise, you need to go `InsertBuffer` pass and check the default value.